src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DefaultProjectSnapshotManager.cs

```diff
@@ -90,12 +90,7 @@ public override IReadOnlyList<ProjectSnapshot> Projects
                 var projects = new ProjectSnapshot[_projects.Count];
                 foreach (var entry in _projects)
                 {
-                    if (entry.Value.Snapshot == null)
-                    {
-                        entry.Value.Snapshot = new DefaultProjectSnapshot(entry.Value.State);
-                    }
-
-                    projects[i++] = entry.Value.Snapshot;
+                    projects[i++] = entry.Value.GetSnapshot();
```

Comment:
The changes here are because I added old and new `ProjectSnapshot` to the changed event. I need this because when a document is removed, I need to be able to see if it *was* an import, and if it *used to* affect some documents' codegen. This is an answer that only the old `ProjectSnapshot` has.

Passing the snapshots in the event simplified some other code as well.